src/main/java/mate/academy/hw01/deepClone/Main.java

```diff
@@ -0,0 +1,97 @@
+package mate.academy.hw01.deepClone;
+
+import java.util.Objects;
+
+public class Main {
+    public static void main(String[] args) throws Exception {
+        Animals myAnimals = new Animals();
+        myAnimals.name = "Cat";
+        Dog newObject = new Dog();
+        newObject.age = 3;
+        myAnimals.newObject = newObject;
+
+        Animals myObject2 = myAnimals.clone();
+        System.out.println("myObject2.name = " + myObject2.name);
+        System.out.println("myObject2.age = " + myObject2.newObject.age);
+        System.out.println(myAnimals.hashCode() == myObject2.hashCode());
+
+        System.out.println(myAnimals.name);
+
+        myObject2.newObject.age = 3;
+        System.out.println("myObject2.age = " + myObject2.newObject.age);
+        System.out.println("myAnimals.age = " + myAnimals.newObject.age + "\n");
+
+        System.out.println(myAnimals != myObject2);
+        System.out.println(myAnimals.getClass() == myObject2.getClass());
+        System.out.println(myAnimals.equals(myObject2));
+
+    }
+}
+
+class Animals implements Cloneable {
+    String name;
+    Dog newObject;
+
+    @Override
+    protected Animals clone() {
+        Animals animals = null;
+        try {
+            animals = (Animals) super.clone();
+            animals.newObject = newObject.clone();
+        } catch (CloneNotSupportedException e) {
```

Comment:
In case of CloneNotSupportedException  you will return null. This is not what user of clone expect. Do not handle exceptions here